### PR TITLE
Update book language picker

### PR DESCRIPTION
**Pull request by jasonlong**
Before committing this one right to #1179, I wanted to make sure it had a little more visibility since it's not just styling changes. 

As part of the book page redesign (#1185), this puts the language options in a dropdown at the top of the sidebar.

image

image

Selecting a different language redirects to the top-level book page for that language.

### CODE REVIEW

File: lib/tasks/book2.rake [hunk 1]
```diff
@@ -32,7 +32,21 @@ end
 def genbook(code, &get_content)
   template_dir = File.join(Rails.root, "templates")
 
-  nav = '<div id="nav"><a href="[[nav-prev]]">prev</a> | <a href="[[nav-next]]">next</a></div>'
+  nav = '
```
Comment (jasonlong) [inline lib/tasks/book2.rake]: Oops, I didn't mean for this file to be in this PR, but it'll soon be in #1179 anyhow.

**Comment by jasonlong** [comment]
To address #1178, I could add a "Want to help with this translation?" link below the dropdown.

File: app/views/books/_translations.html.erb [hunk 2]
```diff
@@ -1,62 +1,39 @@
...
-  <tr><td><a href="/book/tr">Türkçe</a>,</td></tr>
-<tr><td><a href="/book/uz/v2">Ўзбекча</a>.</td></tr>
-</table>
-</p>
-<hr class="sidebar"/>
-<p>
-The source of this book is <a href="https://github.com/progit/progit2">hosted on GitHub.</a></br>
-Patches, suggestions and comments are welcome.
-</p>
+<form class="js-book-lang pv2 " method="get" action="#">
```
Comment (pedrorijo91) [inline app/views/books/_translations.html.erb]: this list needs to be templated instead of hardcoded right?

**Comment by pedrorijo91** [comment]
failed to create the deploy app:

```
-----> Node.js app detected
       
-----> Creating runtime environment
       
       NPM_CONFIG_LOGLEVEL=error
       NODE_ENV=production
       NODE_MODULES_CACHE=true
       NODE_VERBOSE=false
       
-----> Installing binaries
       engines.node (package.json):  unspecified
       engines.npm (package.json):   unspecified (use default)
       engines.yarn (package.json):  unspecified (use default)
       
       Resolving node version 10.x...
       Downloading and installing node 10.13.0...
       Using default npm version: 6.4.1
       Resolving yarn version 1.x...
       Downloading and installing yarn (1.12.3)...
       Installed yarn 1.12.3
       
-----> Building dependencies
       Installing node modules (yarn.lock)
       yarn install v1.12.3
       [1/4] Resolving packages...
       warning webpack-cli > jscodeshift > babel-preset-es2015@6.24.1: 🙌  Thanks for using Babel: we recommend using babel-preset-env now: please read babeljs.io/env to update! 
       warning webpack-cli > webpack-addons > jscodeshift > babel-preset-es2015@6.24.1: 🙌  Thanks for using Babel: we recommend using babel-preset-env now: please read babeljs.io/env to update! 
       warning webpack-cli > jscodeshift > nomnom@1.8.1: Package no longer supported. Contact support@npmjs.com for more info.
       warning webpack-cli > webpack-addons > jscodeshift > nomnom@1.8.1: Package no longer supported. Contact support@npmjs.com for more info.
       error Your lockfile needs to be updated, but yarn was run with `--frozen-lockfile`.
       info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
-----> Build failed
 !     Outdated Yarn lockfile
       Your application contains a Yarn lockfile (yarn.lock) which does not
       match the dependencies in package.json. This can happen if you use npm
       to install or update a dependency instead of Yarn.
       Please run the following command in your application directory and check
       in the new yarn.lock file:
       $ yarn install
       $ git add yarn.lock
       $ git commit -m "Updated Yarn lockfile"
       $ git push heroku master
    
       https://kb.heroku.com/why-is-my-node-js-build-failing-because-of-an-outdated-yarn-lockfile
 !     Push rejected, failed to compile Node.js app.
 !     Push failed
```

File: app/views/books/_translations.html.erb (hunk 2 above)
Comment (jasonlong) [reply app/views/books/_translations.html.erb]: Not sure I follow. The list is already hardcoded in the partial. I just moved each option into this `select` dropdown.

Comment (pedrorijo91) [reply app/views/books/_translations.html.erb]: oh, right.  overlooked. I don't see any reason for not using the info from the DB. but we can add an issue to see that afterwards

Comment (jasonlong) [reply app/views/books/_translations.html.erb]: I'm hoping we can :fire: the database at some point in the future, so I'm all for not relying on it more than we already are.

**Comment by jasonlong** [comment]
Yeah, I need to figure out what's going with yarn for me locally. I'll get that ironed out.

**Comment by pedrorijo91** [comment]
let me know if you want me to push a commit solving the yarn issue for this branch. I basically did a `yarn install`

**Comment by jasonlong** [comment]
Yeah, that's what I'm doing locally but for some reason, it's updating the lock file with older versions of a bunch of packages?

File: app/views/books/_translations.html.erb (hunk 2 above)
Comment (peff) [reply app/views/books/_translations.html.erb]: Yeah, I think it is OK to just keep it in the template source, as long as there is only one such location.

**Comment by peff** [comment]
I think the yarn issues are because I did an `npm update` on the master branch (which does not have a `yarn.lock` yet) to fix a few security issues. On these v2 branches, the resulting `package-lock.json` has been removed (e.g., via b7023165771727abf0ac575bb6f84c20a440784a) and `yarn.lock` needs to be updated instead. So yeah, a `yarn install` on the branch will fix it (e.g., d194ca69c3807f3d8a2774496a4266b35a501e56 which I pushed to the `v2` branch).

If it's easier, we could move to `yarn` on the `master` branch, too. I don't even really know what it is, or why we would care between the two. AFAIK all of the node stuff is just to support linting, etc.

**Comment by jasonlong** [comment]
> Just noticed your comment that the landing page hasn't been touched yet. So probably you were well aware of this problem already.

Yep, I think this will be the main place where users switch their language so those options will certainly be included there.

> if you're on a page and switch languages, it just takes you to the main landing page for that language. 

Agreed that it would be a much nicer UX if the same page reloaded, but as you noticed this doesn't seem feasible currently. At least for now, the feature will be functionally equivalent to what we have now and that's something that could be improved in the future independently.

> changing the select box doesn't do anything unless you have javascript enabled; should we have a submit button as a fallback option?

I'm going to make a note to explore using native `<summary>`/`<details>` controls to implement a similar UI w/o JS.

> Is it worth marking the select box with some text like "Language:"?

I'll think about this w/ the above `<summary>`/`<details>` work.

**Pull request merged**